app/core/exceptions/user.py

```diff
@@ -1,7 +1,7 @@
 from .base import CustomException
 
 class UserAlreadyExistsException(CustomException):
-    code = 409
+    code = 400
```

Comment:
I prefer using 409 here, but if you want to change it to 400 I think that's fine we should just delineate between `UserAlreadyExistsException` and `DataConflictException` or something like that. The `DataConflictException` would be the one that returns 409 and would be returned if the client tries to _change_ a resource in a way that's incompatible, such as updating a submission to be open before the published date, or once we have a job queue, I think it'd be useful to return a 409 if the client tries to submit another request for a job where one is already in the queue (and only one is allowed at a time).

But my vote is just to do all of the above and also keep this to be a 409 because a `UserAlreadyExistsException` to me feels like a subclass of the `DataConflictException`.

Reply:
In general I think using specific error codes for (correctly applicable) specific scenarios is a good thing so that the client can respond differently to the different error codes. It helps the client respond differently if they know they got a 409 than if they got just a 400 for some other reason (which probably indicates a malformed request, not a data conflict).

Reply:
I think I saw a lot of 400s elsewhere in user exceptiosns and wanted to make it exceptions, but honestly 409 is probably more appropriate. I need to review our 400 vs 409 usage a bit more thoroughly, since although interchangeable don’t serve the same purpose semantically

Reply:
In reality, if I understand the nuance between 400 and 409, there really aren't any custom exceptions we should a 400 over 409 for. They should all change from 400 to 409 actually. That being said, I don't fully understand the nuance between them. I've always gotten conflicting information about their usage.